jbpm-container-test/jbpm-container-test-suite/src/test/java/org/jbpm/test/container/archive/EJBService.java

```diff
@@ -0,0 +1,359 @@
+import org.jbpm.test.container.JbpmContainerTest;
+import org.jbpm.test.container.tools.IntegrationMavenResolver;
+import org.jboss.shrinkwrap.api.ArchivePaths;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.EmptyAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.resolver.api.maven.PomEquippedResolveStage;
+import org.jbpm.kie.services.impl.KModuleDeploymentUnit;
+import org.jbpm.services.api.DeploymentService;
+import org.jbpm.services.api.ProcessService;
+import org.jbpm.services.api.model.DeploymentUnit;
+import org.kie.api.KieServices;
+import org.kie.api.builder.KieBuilder;
+import org.kie.api.builder.KieFileSystem;
+import org.kie.api.builder.Message;
+import org.kie.api.builder.ReleaseId;
+import org.kie.api.builder.model.KieBaseModel;
+import org.kie.api.builder.model.KieModuleModel;
+import org.kie.api.builder.model.KieSessionModel;
+import org.kie.api.conf.EqualityBehaviorOption;
+import org.kie.api.conf.EventProcessingOption;
+import org.kie.api.io.Resource;
+import org.kie.api.runtime.conf.ClockTypeOption;
+import org.kie.internal.io.ResourceFactory;
+import org.kie.scanner.MavenRepository;
+import org.kie.scanner.embedder.MavenSettings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class EJBService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(EJBService.class);
+
+    public static final String NAME = "ejb-services";
+
+    public static final String SERVICE_URL = "http://localhost:" + System.getProperty("container.port") +
+            "/" + NAME + "/";
+
+    public static final String EJB_SERVICES_PACKAGE = "org.jbpm.test.container.archive.ejbservices";
+    public static final String TEST_EJB_SERVICES_PACKAGE = "org.jbpm.test.container.test.ejbservices";
+    public static final String EJB_SERVICES_ASSETS_PATH = "/org/jbpm/test/container/archive/ejbservices/assets/";
+
+    protected List<DeploymentUnit> units = new ArrayList<DeploymentUnit>();
+
+    protected List<Long> pids = new ArrayList<Long>();
+
+    private WebArchive war;
+
+    private DeploymentService deploymentService;
+
+    private ProcessService processService;
+
+    public WebArchive buildArchive() throws Exception {
+        PomEquippedResolveStage resolver = IntegrationMavenResolver.get("jbpm",
+                "jbpm-ejb-services", "jbpm-persistence", "issue-keeper");
+
+        System.out.println("### Building archive ...");
+
+        // Remove from dependencies:
+        // 1/ hibernate-jpa-*.jar - causes EJB deployment issues (incompatibility between classes)
+        List<File> minimalDependencies = new ArrayList<File>();
+        File[] dependencies = resolver.importCompileAndRuntimeDependencies().resolve().withTransitivity().asFile();
+        for (File d : dependencies) {
+            System.out.println(d.getName());
+            if(!d.getName().startsWith("hibernate-jpa-") && !d.getName().startsWith("cdi-api-")
```

Comment:
This is difficult to read; you could use separate class for filtering dependencies. Or maybe even sort out your dependencies in pom.xml you use to create the test WAR.